### PR TITLE
fix: 🧪 count and params types

### PR DESCRIPTION
**Pull request by Beraliv**
Summary:

- [x] fixed types for `count` and `params`
- [x] added type tests for `count`, `params` and their combination (wrap each other)
- [x] fix current tests

### CODE REVIEW

File: count/index.type.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+const f2 = count(
+  params<{ category: number }>({
+    one: 'One page in {category}',
+    many: '{count} pages in {category}'
+  })
+)
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+type cases = [
```
Comment (ai) [inline count/index.type.ts:22]: Cool tool!

File: count/index.type.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { AssertTrue as Assert, IsExact } from 'conditional-type-checks'
```
Comment (ai) [inline count/index.type.ts:1]: I use `types.ts` for type tests

Comment (Beraliv) [reply count/index.type.ts:1]: I will move it there

Comment (Beraliv) [reply count/index.type.ts]: Done ✅

**Comment by Beraliv** [comment]
@ai sorry for a late response

seems like I fixed all the problems for now

File: create-i18n/index.d.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ export interface ComponentsJSON {
 
 export interface TranslationFunction<
   Arguments extends any[] = any[],
-  Output extends TranslationJSON = TranslationJSON
+  Output = TranslationJSON | Translation
```
Comment (Beraliv) [inline create-i18n/index.d.ts:18]: added `Translation` here to be able to have a combination of `count` and `params`

**Comment by ai** [comment]
Can I ask to resolve conflicts? I'm afraid to miss something important.

**Comment by Beraliv** [comment]
> Can I ask to resolve conflicts? I'm afraid to miss something important.

Sorry for the late response, was relocating to a different country, let me resolve it and I will 🏓 ping ⏰ you once it's ready

**Comment by Beraliv** [comment]
Ah, okay, this change was included in https://github.com/nanostores/i18n/pull/14 so we can just close it

@ai wdyt?

**Pull request closed without merging**